lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -456,8 +485,9 @@ class AwsCompileFunctions {
     const allFunctions = this.serverless.service.getAllFunctions();
     return BbPromise.each(
       allFunctions,
-      functionName => this.compileFunction(functionName)
-    );
+      functionName => this.downloadS3Artifact(functionName)
```

Comment:
Maybe we can simplify this somehow?

Reply:
Logicaly I think it'll be cleaner if setup as:

```javascript
if (isPackagedExternally) return this.downloadFunctionArtifact(function);
return this.compileFunction(function);
```

Reply:
I like the separation. I've moved the `downloadPackageArtifacts` to the "root" Promise chain of `AwsCompileFunctions`. This way both functions (downloading the package and compiling the functions) are separate.